recipes/mp-units/2.2.0/conanfile.py

```diff
@@ -0,0 +1,267 @@
+        compiler = self.settings.compiler
+        opt = self.options.get_safe(name)
+        feature_name = self._option_feature_map[name]
+        feature = self._feature_compatibility[feature_name]
+        min_version = feature["compiler"].get(str(compiler))
+        return bool(
+            opt == True
+            or (
+                opt == "auto"
+                and min_version
+                and not loose_lt_semver(str(compiler.version), min_version)
+            )
+        )
+
+    @property
+    def _build_cxx_modules(self):
+        return self._is_feature_enabled("cxx_modules")
+
+    @property
+    def _use_fmtlib(self):
+        return not self._is_feature_enabled("std_format")
+
+    def configure(self):
+        if self.options.freestanding:
+            self.options.rm_safe("std_format")
+
+    def requirements(self):
+        if not self.options.freestanding:
+            if self.options.contracts == "gsl-lite":
+                self.requires("gsl-lite/0.41.0")
+            elif self.options.contracts == "ms-gsl":
+                self.requires("ms-gsl/4.0.0")
+            if self._use_fmtlib:
+                self.requires("fmt/10.2.1")
+
+    def validate(self):
+        self._check_feature_supported("mp-units", "minimum_support")
+        for key, value in self._option_feature_map.items():
+            if self.options.get_safe(key) == True:
+                self._check_feature_supported(key, value)
+        if self.options.freestanding and self.options.contracts != "none":
+            raise ConanInvalidConfiguration(
+                "'contracts' should be set to 'none' for a freestanding build"
+            )
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self._build_cxx_modules:
+            tc.cache_variables["CMAKE_CXX_SCAN_FOR_MODULES"] = True
+            tc.cache_variables["MP_UNITS_BUILD_CXX_MODULES"] = str(
+                self.options.cxx_modules
+            ).upper()
+        if self.options.freestanding:
+            tc.cache_variables["MP_UNITS_API_FREESTANDING"] = True
+        else:
+            tc.cache_variables["MP_UNITS_API_STD_FORMAT"] = str(
+                self.options.std_format
+            ).upper()
+        tc.cache_variables["MP_UNITS_API_STRING_VIEW_RET"] = str(
+            self.options.string_view_ret
+        ).upper()
+        tc.cache_variables["MP_UNITS_API_NO_CRTP"] = str(self.options.no_crtp).upper()
+        tc.cache_variables["MP_UNITS_API_CONTRACTS"] = str(
+            self.options.contracts
+        ).upper()
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder="src")
+        if self._build_cxx_modules:
+            cmake.build()
```

Comment:
if having `_build_cxx_modules` cause the _contents_ of the package to be different, we can no longer do `self.info.clear()` - if it doesnt cause the contents of the package to be different, I'm assuming this still doesn't do much (the consumer would still have to build the module interface units on their end anyway, for which Conan doesn't currently provide any support) - not majorly concerned, considering the current limitations of modules - but may be worth issuing a warning when `_build_cxx_modules` is true

Reply:
Yeah, this was my concern here. I am really not sure what lands in the MIU and how it is dependent on the configuration branches. There are preprocessor ifdefs in the code that provide different interfaces based on Conan options.

Reply:
> but may be worth issuing a warning when _build_cxx_modules is true

What kind of warning would you like to see it? "It probably does not work and you are on your own"? 😂